packages/medusa/src/api-v2/admin/stock-locations/[id]/route.ts

```diff
@@ -0,0 +1,31 @@
+import {
+  ContainerRegistrationKeys,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
+import { MedusaRequest, MedusaResponse } from "../../../../types/routing"
+
+import { MedusaError } from "@medusajs/utils"
+
+export const GET = async (req: MedusaRequest, res: MedusaResponse) => {
+  const { id } = req.params
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+
+  const [stock_location] = await remoteQuery(
```

Comment:
comment: @carlos-r-l-rodrigues, I think your idea of adding a "throw-if-not-found" mechanism to remote query could make sense. I think in most endpoints, we are not checking for the existence of the retrieved resource when doing a get by ID.

Reply:
Or alternatively, we add the `if (!result) {` to all get one endpoints, but I think having this baked in would be nice.